Source/AliveLibAO/Abe.cpp

```diff
@@ -2039,10 +2040,107 @@ void Abe::FallOnBombs_4231B0()
     }
 }
 
+__int16 Abe::CrouchOrPushWall(const FP gridSize)
+{
+    if (!WallHit_401930(field_BC_sprite_scale * FP_FromInteger(50), gridSize))
+    {
+        return 1;
+    }
+
+    if (WallHit_401930(field_BC_sprite_scale * FP_FromInteger(20), gridSize))
+    {
+        field_B4_velx = FP_FromInteger(0);
+        field_FC_current_motion = eAbeStates::State_72_PushWall_4292A0;
+        Abe_SFX_2_42A220(9u, 0, 0x7FFF, this);
+        return 0;
+    }
+    else
+    {
+        field_FC_current_motion = eAbeStates::State_21_StandToCrouch_427F40;
+        return 1;
+    }
+}
+
 __int16 Abe::ToLeftRightMovement_422AA0()
 {
-    NOT_IMPLEMENTED();
-    return 0;
+    field_B8_vely = FP_FromInteger(0);
```

Comment:
maybe check AE to see if the logic can be re-arranged a bit? Seems like flipX and some other stuff can be grouped a bit better for readablity? If not I can re-arrange it later